### PR TITLE
feat(event): #MA-950 add filter for lateness and followed abs in export

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
The csv and pdf exports did not correctly filter the events. I had to add the filter on delays without a reason for the csv and the pdf, as well as the filters for absence followed and regularize.

Added PowerMockito for testing.

Doc: https://entconf.gdapublic.fr/display/MP/Event#Event-ExportCSV/PDF

## Checklist tests
Go to the list of events and filter it, export CSV and PDF.

## Issue ticket number and link
https://entsupport.gdapublic.fr/browse/MA-950

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [x] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: presences/src/main/java/fr/openent/presences/service/impl/DefaultEventService.java [hunk 1]
```diff
@@ -342,7 +342,7 @@ private String getEventsQueryPagination(String structureId, String startDate, St
     }
 
     private String setParamsForQueryEvents(List<String> listReasonIds, List<String> userId, Boolean regularized,
```
Comment (BastienBenard) [inline presences/src/main/java/fr/openent/presences/service/impl/DefaultEventService.java:344]: https://pastebin.com/HPGLnur6

File: presences/src/main/java/fr/openent/presences/service/impl/DefaultExportEventService.java [hunk 2]
```diff
@@ -111,7 +111,8 @@ public Future<List<Event>> getCsvData(String structureId, String startDate, Stri
                                           List<String> classes, Boolean regularized, Boolean followed) {
         Promise<List<Event>> promise = Promise.promise();
 
-        getCsvData(structureId, startDate, endDate, eventType, listReasonIds, noReason, userId, userIdFromClasses,
+        //todo Must be change before merge
```
Comment (JinDouang) [inline presences/src/main/java/fr/openent/presences/service/impl/DefaultExportEventService.java]: il sera enlevé après la PR ?

**Comment by JinDouang** [comment]
Summary of all failing tests
FAIL presences/src/main/resources/public/ts/services/__tests__/event.service.test.ts
  ● EventService › check export methode in EventService

    expect(received).toEqual(expected) // deep equality

    Expected: "/presences/events/export?type=CSV&structureId=structureId&startDate=2000-01-02&endDate=2000-02-01&noReason=true&eventType=4,5&reasonIds=1,2,3&userId=userId&classes=classes1,classes2&regularized=true"
    Received: "/presences/events/export?type=CSV&structureId=structureId&startDate=2000-01-02&endDate=2000-02-01&noReason=true&noReasonLateness=true&eventType=4,5&reasonIds=1,2,3&userId=userId&classes=classes1,classes2&regularized=true"

File: build.gradle [hunk 3]
```diff
@@ -66,6 +66,9 @@ subprojects {
         testCompile "org.junit.jupiter:junit-jupiter-engine:$junitVersion"
         testCompile "org.mockito:mockito-core:$mockitoVersion"
         testCompile "org.powermock:powermock-core:$powerMockVersion"
+        testCompile "org.powermock:powermock-api-mockito2:$powerMockVersion"
+        testCompile "org.powermock:powermock-module-junit4:$powerMockVersion"
+        testCompile "org.powermock:powermock-module-testng:$powerMockVersion"
```
Comment (BastienBenard) [inline build.gradle:71]: Utilisé pour les testes

**Review by JinDouang — APPROVED** [review]

File: presences/src/main/java/fr/openent/presences/service/EventService.java [hunk 4]
```diff
@@ -192,13 +192,13 @@ void getEventsByStudent(Integer eventType, List<String> students, String structu
 
     Future<JsonArray> getEventsByStudent(Boolean canSeeAllStudent, Integer eventType, List<String> students, String structure, Boolean justified,
```
Comment (flmartineau) [inline presences/src/main/java/fr/openent/presences/service/EventService.java]: Vu que tu changes les paramètres, tu es obligé de garder le paramètre `justified` sur ces méthodes ? Il me semble qu'il ne sert plus à rien.

**Review by qpic-cgi — APPROVED** [review]

**Pull request merged**